canonical-web-and-design/maas-ui
# fix(legacy): prevent removal of default commissioning scripts from machine details form

## Done

- The third-party ngTagsInput library doesn't allow disabling a subset of tags, so I've used a terrible css hack to prevent default commissioning scripts from being removed in the legacy form.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine details page of a machine that can be commissioned and open the Commission form
- Check that the default commissioning scripts show up by default, they cannot be removed and they are sorted alphanumerically
- Add a non-default script and check that you can then remove it
- Open the Test action form and check that it has remained unchanged

## Fixes

Fixes #1668
